src/sparseml/transformers/utils/initializers.py

```diff
@@ -206,20 +206,53 @@ def resolve_recipe_application(
 ) -> Union[str, Path, None]:
     """
     Resolve the recipe to apply to the model.
-    If the recipe is None, will look for a recipe in the model_path
-
     :param recipe: the recipe to apply to the model.
-        If None, will look for a recipe in the model_path
+        It can be one of the following:
+        - None (no recipe will be applied or the
+            default recipe will be applied if exists. Default recipe
+            is assumed to be stored in the model_path and named RECIPE_NAME)
+        - a path to the recipe file
+        - name of the recipe file (e.g. "recipe.yaml")
+            (assumed to be stored in the model_path instead
+            of RECIPE_NAME)
+        - a string containing the recipe
     :param model_path: the path to the model to load
     :return: the resolved recipe
     """
-    default_recipe = os.path.join(model_path, RECIPE_NAME)
-    requested_recipe = None
-    if recipe:
-        requested_recipe = (
-            recipe if os.path.isfile(recipe) else os.path.join(model_path, recipe)
+
+    if recipe is None:
+        # if recipe is None -> still look for recipe.yaml in the model_path
+        recipe = os.path.join(model_path, RECIPE_NAME)
+        if os.path.isfile(recipe):
+            return recipe
+
+    elif os.path.isfile(recipe):
+        # recipe is a path to a recipe file
+        return _resolve_recipe_file(recipe, model_path)
+
+    elif os.path.isfile(os.path.join(model_path, recipe)):
+        # recipe is a name of a recipe file
+        recipe = os.path.join(model_path, recipe)
+        return _resolve_recipe_file(recipe, model_path)
+    elif isinstance(recipe, str):
+        # recipe is a string containing the recipe
+        _LOGGER.debug(
+            "Applying the recipe string directly to the model, without "
+            "checking for a potential existing recipe in the model_path."
         )
+        return recipe
+
+    _LOGGER.info(
```

Comment:
Could we add another log statement on success that prints out the path to the recipe we are loading?